docs/src/reference/compilation.rst

```diff
@@ -429,6 +429,64 @@ running session.  Please check `Sage documentation
 You can tailor the behavior of the Cython compiler by specifying the
 directives below.
 
+.. _compiling_notebook:
+
+Compiling with a Jupyter Notebook
+=================================
+
+It's possible to compile code in a notebook cell with Cython.
+For this you need to load the Cython magic::
+
+    %load_ext Cython
+
+Then you can define a Cython cell by writing ``%%cython`` on top of it.
+Like this::
+
+    %%cython
+
+    cdef int a = 0
+    for i in range(10):
+        a += i
+    print(a)
+
+Note that each cell will be compiled individually. So if you use a package in a Cython
+cell, you will have to import this package in the same cell. It's not enough to
+have imported the package in a previous cell. Cython will tell you about it at
+compilation time if you don't comply.
+
+Additional allowable arguments to the Cython magic are:
+
+============================================  =======================================================================================================================================
+
+-a, --annotate                                Produce a colorized HTML version of the source.
+
+-+, --cplus                                   Output a C++ rather than C file.
+
+-f, --force                                   Force the compilation of a new module, even if the source has been previously compiled.
+
+-3                                            Select Python 3 syntax
+
+-2                                            Select Python 2 syntax
+
+-c=COMPILE_ARGS, --compile-args=COMPILE_ARGS  Extra flags to pass to compiler via the extra_compile_args.
+
+--link-args LINK_ARGS                         Extra flags to pass to linker via the extra_link_args.
+
+-l LIB, --lib LIB                             Add a library to link the extension against (can be specified multiple times).
+
+-L dir                                        Add a path to the list of libary directories (can be specified multiple times).
+
+-I INCLUDE, --include INCLUDE                 Add a path to the list of include directories (can be specified multiple times).
+
+-S, --src                                     Add a path to the list of src files (can be specified multiple times).
+
+-n NAME, --name NAME                          Specify a name for the Cython module.
+
+--pgo                                         Enable profile guided optimisation in the C compiler. Compiles the cell twice and executes it in between to generate a runtime profile.
+
+--verbose                                     Print debug information like generated .c/.cpp file location and exact gcc/g++ command invoked.
+============================================  =======================================================================================================================================
```

Comment:
Just an idea, and I'm not sure it's worth the effort, but I think it would be nice if this option list could be generated from the docstring in `IpythonMagic.py` somehow. There's a test setup in `Cython/Build/Tests/TestIpythonMagic.py` that sets up and runs IPython, and there might be a way to use something similar to output the cell magic documentation (and then commit it into an include file in `docs`).

OTOH, maybe we should (also? only?) include a comment on how to look up the documentation of the cell magic in IPython... Its help system seems to call this internally: `get_ipython().magic('pinfo cython')`.

Reply:
I suggest:

- We keep this ugly copy pasting as a v1.0, so that the users have access to it quickly.
- In the future, if I have some time to spare, I can look into importing those descriptions automatically as you suggest (I think there are more urgent things to do in the documentation, but I'm also not fond of copy-pasting, it's not very maintanable, and would give more work to do in the long run).
- As a bonus, we give the users the information on how to find those descriptions from IPython. But we leave the descriptions here also. I'm not very found of making the users work to access information. It defeats the purpose of the documentation.

Reply:
Sure, works for me.